### ISSUE TITLE
capr4n theme: Cursor jump to to beginning of line and overwrite promt.

### ISSUE THREAD

**Issue by tanpro260196** [JanDeDobbeleer/oh-my-posh#5221]
### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

### What happened?

After update to 21.21, Using capr4n theme will cause the cursor to jump to beginning of line (before the promt) and overwrite everything from there.
![image](https://github.com/JanDeDobbeleer/oh-my-posh/assets/12226439/46ad55ea-04ea-4984-95cd-d4495b25f09d)

I also cannot press enter to create new line.


### Theme

capr4n

### What OS are you seeing the problem on?

Linux

### Which shell are you using?

bash

### Log output

```Shell
Version: 21.21.1

Shell: bash (5.1.16(1)-release)

Prompt:

 xxx  ~                                                                                                                                                                                                                                                    18:25:08   0ms

Segments:

ConsoleTitle(true)                         -   0 ms
Time(true)                                 -   0 ms
Executiontime(true)                        -   0 ms
Session(true)                              -   0 ms
Path(true)                                 -   0 ms

Run duration: 9.550899ms

Cache path: /home/xxx/.cache/oh-my-posh

Config path: /home/xxx/capr4n.omp.json

Logs:

[DEBUG] 18:25:08.376 os.go:Getenv:306 → NO DATA
[TRACE] 18:25:08.376 os.go:Getenv(OMP_CACHE_DIR) - 95.898µs
[DEBUG] 18:25:08.376 os.go:Getenv:306 → NO DATA
[TRACE] 18:25:08.376 os.go:Getenv(XDG_CACHE_HOME) - 7.051µs
[TRACE] 18:25:08.376 os_unix.go:CachePath() - 164.977µs
[DEBUG] 18:25:08.377 os.go:Getenv:306 → /home/xxx/capr4n.omp.json
[TRACE] 18:25:08.377 os.go:Getenv(POSH_THEME) - 9.965µs
[DEBUG] 18:25:08.377 os_unix.go:Platform:87 → ubuntu
[TRACE] 18:25:08.377 os.go:resolveConfigPath() - 23.482µs
[TRACE] 18:25:08.377 os.go:Init() - 444.833µs
[TRACE] 18:25:08.377 os.go:Flags() - 80ns
[TRACE] 18:25:08.378 load.go:loadConfig() - 1.580502ms
[TRACE] 18:25:08.378 os.go:Flags() - 73ns
[DEBUG] 18:25:08.378 debug.go:44 → Terminal shell: %s shell
[DEBUG] 18:25:08.378 debug.go:44 → Terminal program: %s Unknown
[DEBUG] 18:25:08.378 os.go:Getenv:306 → NO DATA
[TRACE] 18:25:08.378 os.go:Getenv(OMP_CACHE_DISABLED) - 4.92µs
[DEBUG] 18:25:08.378 os.go:Shell:581 → no shell name provided in flags, trying to detect it
[DEBUG] 18:25:08.380 os.go:Shell:589 → process name: bash
[TRACE] 18:25:08.380 os.go:Shell() - 1.850675ms
[DEBUG] 18:25:08.380 os.go:Getenv:306 → 5.1.16(1)-release
[TRACE] 18:25:08.380 os.go:Getenv(POSH_SHELL_VERSION) - 6.368µs
[DEBUG] 18:25:08.380 debug.go:PrintDebug:23 → Segment: Title
[DEBUG] 18:25:08.380 text.go:Render:73 → Rendering template: {{if .Root}} ⚡ {{end}}{{.UserName}} ➔ 📁{{.Folder}}
[TRACE] 18:25:08.381 os_unix.go:Root() - 4.466µs
[TRACE] 18:25:08.381 os.go:Shell() - 121ns
[TRACE] 18:25:08.381 os.go:StatusCodes() - 73ns
[DEBUG] 18:25:08.381 os_unix.go:IsWsl:35 → false
[TRACE] 18:25:08.381 os_unix.go:IsWsl() - 11.256µs
[DEBUG] 18:25:08.381 os.go:TemplateCache:779 → environment: [SHELL=/bin/bash POSH_PID=2212868 LANGUAGE=en_CA:en PWD=/home/xxx LOGNAME=xxx XDG_SESSION_TYPE=tty MOTD_SHOWN=pam HOME=/home/xxx LANG=en_CA.UTF-8 LS_COLORS=rs=0:di=01;34:ln=01;36:mh=00:pi=40;33:so=01;35:do=01;35:bd=40;33;01:cd=40;33;01:or=40;31;01:mi=00:su=37;41:sg=30;43:ca=30;41:tw=30;42:ow=34;42:st=37;44:ex=01;32:*.tar=01;31:*.tgz=01;31:*.arc=01;31:*.arj=01;31:*.taz=01;31:*.lha=01;31:*.lz4=01;31:*.lzh=01;31:*.lzma=01;31:*.tlz=01;31:*.txz=01;31:*.tzo=01;31:*.t7z=01;31:*.zip=01;31:*.z=01;31:*.dz=01;31:*.gz=01;31:*.lrz=01;31:*.lz=01;31:*.lzo=01;31:*.xz=01;31:*.zst=01;31:*.tzst=01;31:*.bz2=01;31:*.bz=01;31:*.tbz=01;31:*.tbz2=01;31:*.tz=01;31:*.deb=01;31:*.rpm=01;31:*.jar=01;31:*.war=01;31:*.ear=01;31:*.sar=01;31:*.rar=01;31:*.alz=01;31:*.ace=01;31:*.zoo=01;31:*.cpio=01;31:*.7z=01;31:*.rz=01;31:*.cab=01;31:*.wim=01;31:*.swm=01;31:*.dwm=01;31:*.esd=01;31:*.jpg=01;35:*.jpeg=01;35:*.mjpg=01;35:*.mjpeg=01;35:*.gif=01;35:*.bmp=01;35:*.pbm=01;35:*.pgm=01;35:*.ppm=01;35:*.tga=01;35:*.xbm=01;35:*.xpm=01;35:*.tif=01;35:*.tiff=01;35:*.png=01;35:*.svg=01;35:*.svgz=01;35:*.mng=01;35:*.pcx=01;35:*.mov=01;35:*.mpg=01;35:*.mpeg=01;35:*.m2v=01;35:*.mkv=01;35:*.webm=01;35:*.webp=01;35:*.ogm=01;35:*.mp4=01;35:*.m4v=01;35:*.mp4v=01;35:*.vob=01;35:*.qt=01;35:*.nuv=01;35:*.wmv=01;35:*.asf=01;35:*.rm=01;35:*.rmvb=01;35:*.flc=01;35:*.avi=01;35:*.fli=01;35:*.flv=01;35:*.gl=01;35:*.dl=01;35:*.xcf=01;35:*.xwd=01;35:*.yuv=01;35:*.cgm=01;35:*.emf=01;35:*.ogv=01;35:*.ogx=01;35:*.aac=00;36:*.au=00;36:*.flac=00;36:*.m4a=00;36:*.mid=00;36:*.midi=00;36:*.mka=00;36:*.mp3=00;36:*.mpc=00;36:*.ogg=00;36:*.ra=00;36:*.wav=00;36:*.oga=00;36:*.opus=00;36:*.spx=00;36:*.xspf=00;36: POSH_SHELL_VERSION=5.1.16(1)-release CONDA_PROMPT_MODIFIER=false SSH_CONNECTION=192.168.0.223 51295 192.168.0.208 22 LESSCLOSE=/usr/bin/lesspipe %s %s XDG_SESSION_CLASS=user TERM=xterm-256color LESSOPEN=| /usr/bin/lesspipe %s USER=xxx SHLVL=1 XDG_SESSION_ID=6678 POSH_THEME=/home/xxx/capr4n.omp.json XDG_RUNTIME_DIR=/run/user/1000 SSH_CLIENT=192.168.0.223 51295 22 XDG_DATA_DIRS=/usr/share/gnome:/home/xxx/.local/share/flatpak/exports/share:/var/lib/flatpak/exports/share:/usr/local/share:/usr/share:/var/lib/snapd/desktop PATH=/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin:/usr/games:/usr/local/games:/snap/bin DBUS_SESSION_BUS_ADDRESS=unix:path=/run/user/1000/bus SSH_TTY=/dev/pts/0 POWERLINE_COMMAND=oh-my-posh _=/usr/local/bin/oh-my-posh]
[TRACE] 18:25:08.381 os.go:GOOS() - 110ns
[DEBUG] 18:25:08.381 os.go:Pwd:336 → /home/xxx
[TRACE] 18:25:08.381 os.go:Pwd() - 24.963µs
[DEBUG] 18:25:08.381 os_unix.go:IsWsl:35 → false
[TRACE] 18:25:08.381 os_unix.go:IsWsl() - 5.337µs
[TRACE] 18:25:08.381 os.go:GOOS() - 58ns
[TRACE] 18:25:08.381 os.go:GOOS() - 59ns
[TRACE] 18:25:08.381 os.go:GOOS() - 60ns
[TRACE] 18:25:08.381 os.go:GOOS() - 58ns
[TRACE] 18:25:08.381 os.go:GOOS() - 63ns
[TRACE] 18:25:08.381 os.go:GOOS() - 61ns
[TRACE] 18:25:08.381 os.go:GOOS() - 60ns
[TRACE] 18:25:08.381 os.go:GOOS() - 57ns
[TRACE] 18:25:08.381 os.go:GOOS() - 59ns
[TRACE] 18:25:08.381 os.go:GOOS() - 62ns
[TRACE] 18:25:08.381 os.go:GOOS() - 60ns
[TRACE] 18:25:08.381 os.go:GOOS() - 63ns
[TRACE] 18:25:08.381 os.go:GOOS() - 60ns
[TRACE] 18:25:08.381 os.go:GOOS() - 60ns
[DEBUG] 18:25:08.381 os.go:User:474 → xxx
[TRACE] 18:25:08.381 os.go:User() - 4.865µs
[DEBUG] 18:25:08.381 os.go:Host:491 → Main
[TRACE] 18:25:08.381 os.go:Host() - 8.508µs
[TRACE] 18:25:08.381 os.go:GOOS() - 58ns
[DEBUG] 18:25:08.381 os_unix.go:Platform:87 → ubuntu
[DEBUG] 18:25:08.381 os.go:Getenv:306 → 1
[TRACE] 18:25:08.381 os.go:Getenv(SHLVL) - 6.92µs
[TRACE] 18:25:08.381 os.go:TemplateCache() - 297.498µs
[DEBUG] 18:25:08.381 os.go:Getenv:306 → NO DATA
[TRACE] 18:25:08.381 os.go:Getenv(POSH_CURSOR_LINE) - 6.657µs
[DEBUG] 18:25:08.381 os.go:Getenv:306 → NO DATA
[TRACE] 18:25:08.381 os.go:Getenv(POSH_CURSOR_COLUMN) - 3.794µs
[TRACE] 18:25:08.381 os.go:Flags() - 64ns
[TRACE] 18:25:08.381 os.go:Flags() - 62ns
[TRACE] 18:25:08.381 os.go:Shell() - 104ns
[TRACE] 18:25:08.381 os.go:Shell() - 60ns
[DEBUG] 18:25:08.381 block.go:InitPlain:65 → Terminal shell: %s shell
[DEBUG] 18:25:08.381 block.go:InitPlain:65 → Terminal program: %s Unknown
[DEBUG] 18:25:08.381 os.go:Getenv:306 → NO DATA
[TRACE] 18:25:08.381 os.go:Getenv(OMP_CACHE_DISABLED) - 9.432µs
[TRACE] 18:25:08.381 os.go:Flags() - 102ns
[TRACE] 18:25:08.381 os.go:Pwd() - 190ns
[TRACE] 18:25:08.381 os.go:GOOS() - 67ns
[DEBUG] 18:25:08.381 segment.go:SetEnabled:115 → Segment: Executiontime
[TRACE] 18:25:08.381 os.go:Flags() - 204ns
[DEBUG] 18:25:08.381 executiontime.go:Enabled:57 → always_enabled: true
[TRACE] 18:25:08.381 os.go:ExecutionTime() - 81ns
[TRACE] 18:25:08.381 os.go:Pwd() - 188ns
[DEBUG] 18:25:08.381 executiontime.go:Enabled:59 → threshold: 500.000000
[DEBUG] 18:25:08.381 executiontime.go:Enabled:63 → austin
[TRACE] 18:25:08.381 os.go:GOOS() - 75ns
[TRACE] 18:25:08.381 os.go:TemplateCache() - 205ns
[DEBUG] 18:25:08.381 segment.go:SetEnabled:115 → Segment: Time
[DEBUG] 18:25:08.381 time.go:Enabled:29 → 15:04:05
[TRACE] 18:25:08.381 os.go:TemplateCache() - 233ns
[DEBUG] 18:25:08.381 text.go:Render:73 → Rendering template:   {{ .FormattedMs }}
[DEBUG] 18:25:08.381 text.go:Render:73 → Rendering template:   {{ .CurrentDate | date .Format }}
[TRACE] 18:25:08.382 os.go:TemplateCache() - 401ns
[TRACE] 18:25:08.382 os.go:TemplateCache() - 398ns
[DEBUG] 18:25:08.383 text.go:Render:73 → Rendering template: plain
[DEBUG] 18:25:08.383 text.go:Render:73 → Rendering template: plain
[TRACE] 18:25:08.383 os.go:Shell() - 213ns
[TRACE] 18:25:08.383 os.go:Shell() - 62ns
[TRACE] 18:25:08.383 os.go:Shell() - 61ns
[TRACE] 18:25:08.383 os.go:Shell() - 60ns
[TRACE] 18:25:08.383 os.go:Flags() - 98ns
[TRACE] 18:25:08.383 os.go:Pwd() - 189ns
[TRACE] 18:25:08.383 os.go:GOOS() - 80ns
[DEBUG] 18:25:08.383 segment.go:SetEnabled:115 → Segment: Path
[DEBUG] 18:25:08.384 path.go:124 → display_cygpath: false
[TRACE] 18:25:08.384 os.go:GOOS() - 62ns
[TRACE] 18:25:08.384 os.go:PathSeparator() - 68ns
[TRACE] 18:25:08.384 os.go:Pwd() - 84ns
[TRACE] 18:25:08.384 os.go:Shell() - 113ns
[TRACE] 18:25:08.384 os.go:Shell() - 74ns
[DEBUG] 18:25:08.384 path.go:replaceMappedLocations:517 → mapped_locations_enabled: true
[DEBUG] 18:25:08.384 path.go:replaceMappedLocations:518 → 
[DEBUG] 18:25:08.384 path.go:replaceMappedLocations:519 → 
[TRACE] 18:25:08.384 os.go:GOOS() - 64ns
[DEBUG] 18:25:08.384 path.go:replaceMappedLocations:520 → ~
[TRACE] 18:25:08.384 os.go:Flags() - 111ns
[DEBUG] 18:25:08.384 path.go:replaceMappedLocations:525 → mapped_locations: map[]
[TRACE] 18:25:08.384 os.go:GOOS() - 67ns
[TRACE] 18:25:08.384 os.go:GOOS() - 62ns
[TRACE] 18:25:08.384 os.go:Pwd() - 229ns
[TRACE] 18:25:08.384 os.go:GOOS() - 76ns
[DEBUG] 18:25:08.384 path.go:makeFolderFormatMap:841 → NO DATA
[DEBUG] 18:25:08.384 segment.go:SetEnabled:115 → Segment: Git
[TRACE] 18:25:08.384 os.go:Pwd() - 139ns
[DEBUG] 18:25:08.384 os_unix.go:IsWsl:35 → false
[TRACE] 18:25:08.384 os.go:Flags() - 166ns
[TRACE] 18:25:08.384 os.go:TemplateCache() - 185ns
[TRACE] 18:25:08.384 os_unix.go:IsWsl() - 5.166µs
[TRACE] 18:25:08.384 os_unix.go:IsWsl2() - 11.618µs
[TRACE] 18:25:08.384 os.go:StackCount() - 124ns
[TRACE] 18:25:08.384 os.go:GOOS() - 59ns
[TRACE] 18:25:08.384 os.go:Pwd() - 260ns
[TRACE] 18:25:08.384 os.go:GOOS() - 86ns
[TRACE] 18:25:08.384 os_unix.go:DirIsWritable(/home/xxx) - 12.942µs
[DEBUG] 18:25:08.384 segment.go:SetEnabled:115 → Segment: Session
[TRACE] 18:25:08.384 os.go:TemplateCache() - 168ns
[DEBUG] 18:25:08.384 os.go:Getenv:306 → 192.168.0.xxx 51295 192.168.0.xxx 22
[TRACE] 18:25:08.384 os.go:Getenv(SSH_CONNECTION) - 7.573µs
[TRACE] 18:25:08.384 os.go:TemplateCache() - 213ns
[DEBUG] 18:25:08.384 os.go:CommandPath:533 → /usr/bin/git
[TRACE] 18:25:08.384 os.go:CommandPath(git) - 77.713µs
[TRACE] 18:25:08.384 os.go:HasCommand(git) - 83.424µs
[TRACE] 18:25:08.384 os.go:Pwd() - 134ns
[ERROR] 18:25:08.384 os.go:HasParentFilePath:687 → stat .git: no such file or directory
[TRACE] 18:25:08.384 os.go:HasParentFilePath(.git) - 48.047µs
[DEBUG] 18:25:08.384 git.go:shouldDisplay:284 → fetch_bare_info: false
[DEBUG] 18:25:08.384 text.go:Render:73 → Rendering template: {{if .Root}}  {{.UserName}} {{else}} {{.UserName}} {{end}}
[DEBUG] 18:25:08.384 text.go:Render:73 → Rendering template:  {{ .Path }}
[TRACE] 18:25:08.384 os.go:TemplateCache() - 284ns
[TRACE] 18:25:08.384 os.go:TemplateCache() - 400ns
[DEBUG] 18:25:08.384 text.go:Render:73 → Rendering template: powerline
[DEBUG] 18:25:08.384 text.go:Render:73 → Rendering template: diamond
[DEBUG] 18:25:08.384 text.go:Render:73 → Rendering template: powerline
[TRACE] 18:25:08.384 os.go:Shell() - 208ns
[TRACE] 18:25:08.384 os.go:Shell() - 65ns
[TRACE] 18:25:08.384 os.go:Flags() - 66ns
[TRACE] 18:25:08.384 os.go:Shell() - 60ns
[TRACE] 18:25:08.384 os.go:Shell() - 61ns
[DEBUG] 18:25:08.386 os_unix.go:TerminalWidth:80 → terminal width: 280
[TRACE] 18:25:08.386 os_unix.go:TerminalWidth() - 1.370398ms
[DEBUG] 18:25:08.386 primary.go:Primary:108 → Terminal shell: %s shell
[DEBUG] 18:25:08.386 primary.go:Primary:108 → Terminal program: %s Unknown
[DEBUG] 18:25:08.386 os.go:Getenv:306 → NO DATA
[TRACE] 18:25:08.386 os.go:Getenv(OMP_CACHE_DIR) - 4.044µs
[DEBUG] 18:25:08.386 os.go:Getenv:306 → NO DATA
[TRACE] 18:25:08.386 os.go:Getenv(XDG_CACHE_HOME) - 3.579µs
[TRACE] 18:25:08.386 os_unix.go:CachePath() - 34.655µs
[TRACE] 18:25:08.386 os.go:Flags() - 57ns
```

**Comment by tanpro260196** [comment]
Further note: It's Linux but access via ssh using Windows Terminal.

**Comment by JanDeDobbeleer** [comment]
@tanpro260196 weird, nothing changed in that area, which version did you come from?

**Comment by JanDeDobbeleer** [comment]
@tanpro260196 it got introduced [here](https://github.com/JanDeDobbeleer/oh-my-posh/compare/v21.18.1...v21.18.2), which makes no sense yet.

**Comment by JanDeDobbeleer** [comment]
@tanpro260196 this bug seems to originate from a fix we did that now looks like a patch because we didn't escape things correctly. I fixed the escaping a while back, and now that the original issue was solved, this patch came back with a vengeance. I need to validate on linux first before I feel confident, this is now solved on Windows (Git Bash) and macOS but I want to make sure it works everywhere.
